nxdrive/engine/processor.py

```diff
@@ -549,7 +560,7 @@ def _direct_transfer_end(
         self.dao.remove_state(doc_pair, recursive=recursive)
 
         # Update session then handle the status
-        session = self.dao.get_session(doc_pair.session)
+        session = session or self.dao.get_session(doc_pair.session)
```

Comment:
I think we should always get the session from the database here as it can change many times before attaining `_direct_transfer_end()`

Reply:
Should we remove the `session` argument and always fetch a fresh session object?